### PR TITLE
[api-minor] Change `Font.exportData` to use an explicit white-list of exportable properties, and stop exporting internal/unused properties

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Please refer to the individual commit messages for addition details.*

---

Note that there's *still* a number of potentially very large data-structures, containing e.g. Arrays and Objects, being exported despite being completely unused on the main-thread and/or in the API.
However, since we cannot be sure that they're  completely unused in all custom implementations of the PDF.js library, we probably cannot just remove them without also providing users with a way to still get this information (that we've been *accidentally* exposing for years).

To not unnecessarily delay the changes in this PR, which on their own should already be beneficial in my opinion, I thus decided to post-phone any further clean-up to a separate PR.

### CODE REVIEW

**Comment by timvandermeij** [comment]
The bot should be fixed again.

/botio-windows test

**Comment by Snuffleupagus** [comment]
It seems that the Windows bot have decided to take an extended vacation :-)

Anyway, given that the tests pass on the Linux bot (modulo the Chrome-intermittent ones) and that all tests *obviously* also pass locally on Windows for me; do we really need to wait for the Windows bot to be re-started again?
If this patch was fundamentally broken in some way, I'd have expected lots of errors everywhere...

File: src/core/fonts.js [hunk 1]
```diff
@@ -1258,12 +1301,14 @@ var Font = (function FontClosure() {
...
 
-    exportData: function Font_exportData() {
-      // TODO remove enumerating of the properties, e.g. hardcode exact names.
-      var data = {};
-      for (var i in this) {
-        if (this.hasOwnProperty(i)) {
-          data[i] = this[i];
+    exportData() {
+      const data = Object.create(null);
+      let property, value;
```
Comment (timvandermeij) [inline src/core/fonts.js:1300]: I think we can remove this line if we use `for (const property ...` and `const value = this...` below.

**Comment by timvandermeij** [comment]
Looks good after the comment above is addressed.

File: src/core/fonts.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/core/fonts.js]: Actually, that seems *slightly* slower based on bot run-time hence I'm reverting this change (besides those variables will get reused with the follow-up PR anyway).

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Nice improvements!

**Comment by Snuffleupagus** [comment]
> Nice improvements!

Thanks for the review; the real improvements are PR #11777 though :-)

**Pull request merged**